tests/Single/CompressAssociativeTest.php

```diff
@@ -125,14 +125,14 @@ public function dataProviderForArray(): array
             [
                 [1, 2, 3, 'a' => 11, 'b' => 22, 'c' => 33],
                 ['a', 'b', 0, 2],
-                [0, 2, 'a', 'b'],
-                [1, 3, 11, 22],
+                ['a', 'b', 0, 2],
+                [11, 22, 1, 3],
```

Comment:
I think this is an unexpected and undesirable result which changes the definition of the action and what an iterable is. In [PHP, array's](https://www.php.net/manual/en/language.types.array.php) are not dictionaries/maps like which are found in other languages. It is technically an ordered map (or ordered dictionary). The compress function is expected to filter out specific indexes without changing the order of the elements. The expectation would be the same result, regardless if the input was an array, iterator, or generator, etc.

For reference, Python has a similar data structure: [OrderedDict](https://docs.python.org/3/library/collections.html#collections.OrderedDict).

Reply:
I think performance is more important here than order. The main thing is to get exactly the right elements for the right keys.

Reply:
Hi @Smoren,

Thank you for considering how to improve performance. However, performance doesn't matter if it gets the wrong answer. PHP arrays are an ordered map, so the order is fundamentally part of the data structure. If the order changes, the result does not match the input.

Let me demonstrate with two simple examples:

This first example shows intent, and compress subverting expectations:
```php
$soonToBeSorted = ['a' => 4, 'b' => 5, 'c' => 6, 1, 2, 3];  // [4, 5, 6, 1, 2, 3]
asort($soonToBeSorted);                                     // [1, 2, 3, 4, 5, 6]

foreach (Single::compressAssociative($soonToBeSorted, ['a', 'b', 0, 2]) as $x) {
    print($x . ", ");  // 4, 5, 1, 3
}
```
In this example, clearly a sorted result is expected, but compress changed the order to no longer be sorted.

This second example shows a real scenario of data where order matters:
```php
$chessMoves = [
    'whiteOpening' => 'f3',
    'blackOpening' => 'e6', 
    'whiteMidGame' => 'g4',
    'checkmate'    => 'Qh4'
];

$endGameCompressor = ['checkmate', 'whiteMidGame'];

foreach (Single::compressAssociative($chessMoves, $endGameCompressor) as $move => $notation) {
    print("$move: $notation\n");
}

// checkmate: Qh4
// whiteMidGame: g4
```

In this example, we just want to display the interesting part of the game, but compress changes the order, so you can't make sense of the chess game anymore (a real non-fool's mate would have a longer end game with more moves).

Order is fundamental to PHP arrays as well as iterators and generators. Changing the order would be unexpected and most likely result in user complaints, even if it was documented.

Reply:
OK, you are true. I'm closing this PR.